manifests/init.pp

```diff
@@ -6,18 +6,118 @@
 # @example
 #   include common_events
 class common_events (
-  Optional[String]                                $pe_username   = undef,
-  Optional[Sensitive[String]]                     $pe_password   = undef,
-  Optional[String]                                $pe_token      = undef,
-  Optional[Boolean]                               $disabled      = false,
-  Optional[String]                                $cron_minute   = '*/2',
-  Optional[String]                                $cron_hour     = '*',
-  Optional[String]                                $cron_weekday  = '*',
-  Optional[String]                                $cron_month    = '*',
-  Optional[String]                                $cron_monthday = '*',
-  Optional[String]                                $log_path      = undef,
-  Enum['DEBUG', 'INFO', 'WARN', 'ERROR', 'FATAL'] $log_level     = 'WARN',
-  Enum['NONE', 'DAILY', 'WEEKLY', 'MONTHLY']      $log_rotation  = 'NONE',
+  Optional[String]                                $pe_username      = undef,
+  Optional[Sensitive[String]]                     $pe_password      = undef,
+  Optional[String]                                $pe_token         = undef,
+  Optional[String]                                $pe_console       = 'localhost',
+  Optional[Boolean]                               $disabled         = false,
+  Optional[String]                                $cron_minute      = '*/2',
+  Optional[String]                                $cron_hour        = '*',
+  Optional[String]                                $cron_weekday     = '*',
+  Optional[String]                                $cron_month       = '*',
+  Optional[String]                                $cron_monthday    = '*',
+  Optional[String]                                $log_path         = undef,
+  Optional[String]                                $confdir          = "${settings::confdir}/common_events",
+  Optional[String]                                $logfile_basepath = common_events::base_path($settings::logdir, $common_events::log_path),
+  Optional[String]                                $lockdir_basepath = common_events::base_path($settings::statedir, undef),
+  Optional[Tuple]                                 $lock_dirs        = ["${lockdir_basepath}/common_events", "${lockdir_basepath}/common_events/cache/", "${lockdir_basepath}/common_events/cache/state"],
```

Comment:
This doesn't need to be a class parameter. It can be calculated in the body of the manifest from the `$lockdir_basepath` parameter.